2020-05-06 14:59:44 [scrapy] ERROR: Error while obtaining start requests
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/scrapy/core/engine.py", line 127, in _next_request
    request = next(slot.start_requests)
  File "/home/lalit/.local/lib/python3.8/site-packages/scrapy_splash/middleware.py", line 167, in process_start_requests
    for req in start_requests:
  File "/run/media/lalit/Work/scrapy-googlescholar/topl_project/topl_project/spiders/1.py", line 52, in start_requests
    url=self.url
AttributeError: 'CitSpider' object has no attribute 'url'
2020-05-06 14:59:45 [scrapy] DEBUG: Crawled (200) <GET https://scholar.google.com/scholar?oi=bibs&hl=en&cites=10438299327049367009> (referer: None)
2020-05-06 14:59:45 [scrapy] ERROR: Spider error processing <GET https://scholar.google.com/scholar?oi=bibs&hl=en&cites=10438299327049367009> (referer: None)
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/usr/lib/python3.8/site-packages/scrapy/core/spidermw.py", line 84, in evaluate_iterable
    for r in iterable:
  File "/home/lalit/.local/lib/python3.8/site-packages/scrapy_splash/middleware.py", line 156, in process_spider_output
    for el in result:
  File "/usr/lib/python3.8/site-packages/scrapy/core/spidermw.py", line 84, in evaluate_iterable
    for r in iterable:
  File "/usr/lib/python3.8/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/usr/lib/python3.8/site-packages/scrapy/core/spidermw.py", line 84, in evaluate_iterable
    for r in iterable:
  File "/usr/lib/python3.8/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/usr/lib/python3.8/site-packages/scrapy/core/spidermw.py", line 84, in evaluate_iterable
    for r in iterable:
  File "/usr/lib/python3.8/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/usr/lib/python3.8/site-packages/scrapy/core/spidermw.py", line 84, in evaluate_iterable
    for r in iterable:
  File "/usr/lib/python3.8/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/run/media/lalit/Work/scrapy-googlescholar/topl_project/topl_project/spiders/1.py", line 60, in parse
    "authors": divs.css(".gs_a a::text").extract("0"),
TypeError: getall() takes 1 positional argument but 2 were given
2020-05-06 14:59:45 [scrapy] INFO: Closing spider (finished)
2020-05-06 14:59:45 [scrapy] INFO: Dumping Scrapy stats:
{'downloader/request_bytes': 351,
 'downloader/request_count': 1,
 'downloader/request_method_count/GET': 1,
 'downloader/response_bytes': 36125,
 'downloader/response_count': 1,
 'downloader/response_status_count/200': 1,
 'elapsed_time_seconds': 0.966729,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2020, 5, 6, 9, 29, 45, 351464),
 'log_count/DEBUG': 1,
 'log_count/ERROR': 2,
 'log_count/INFO': 7,
 'memusage/max': 62672896,
 'memusage/startup': 62672896,
 'response_received_count': 1,
 'scheduler/dequeued': 1,
 'scheduler/dequeued/memory': 1,
 'scheduler/enqueued': 1,
 'scheduler/enqueued/memory': 1,
 'spider_exceptions/TypeError': 1,
 'start_time': datetime.datetime(2020, 5, 6, 9, 29, 44, 384735)}
2020-05-06 14:59:45 [scrapy] INFO: Spider closed (finished)
